### PR TITLE
Feature/handle start api

### LINKED ISSUE #436: Handle API limit at the beginning of run
**Is your feature request related to a problem? Please describe.**
`MI` is not able to gather data whenever the API limit is reached at the very beginning of launching the application.  

**Describe the solution you'd like**
Add a check for API limit at the start. If exceeded, sleep the same exact way as is done in `iterator` module.

**LOGS**
Found out this in a recently launched data workflow. You can see that the mi did not even start and the API was exceeded (due to uknown reasons, because it should be handled during the aggregation, which should secure every other following subsequent workflow).
```
[xtuchyna@localhost ~]$ oc logs -c=main mi-analysis-mi-210715120826-dc2b0b39cec836d1-1442632437
Traceback (most recent call last):
  File "srcopsmetrics/cli.py", line 196, in <module>
    cli(auto_envvar_prefix="MI")
  File "/opt/app-root/lib64/python3.8/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/opt/app-root/lib64/python3.8/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/opt/app-root/lib64/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/app-root/lib64/python3.8/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "srcopsmetrics/cli.py", line 151, in cli
    repos.extend(GitHubKnowledge.get_repositories(repository=rep.strip()))
  File "/opt/app-root/src/srcopsmetrics/github_knowledge.py", line 68, in get_repositories
    repos.append(gh.get_repo(repository).full_name)
  File "/opt/app-root/lib64/python3.8/site-packages/github/MainClass.py", line 330, in get_repo
    headers, data = self.__requester.requestJsonAndCheck("GET", url)
  File "/opt/app-root/lib64/python3.8/site-packages/github/Requester.py", line 353, in requestJsonAndCheck
    return self.__check(
  File "/opt/app-root/lib64/python3.8/site-packages/github/Requester.py", line 378, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.RateLimitExceededException: 403 {"message": "API rate limit exceeded for user ID 66362888.", "documentation_url": "https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
```

/kind feature

### PR DESCRIPTION
**Pull request by xtuchyna**
## Related Issues and Dependencies
Fixes #436 

## This introduces a breaking change

- [ ] Yes
- [X] No

## Description
Now, GH API limit is checked also at the start of `mi`.
Also, progress bar was slightly changed - the logging of remaining API rate limit was moved as its postfix.

### CODE REVIEW

File: srcopsmetrics/github_handling.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        """Initialize with github object."""
+        if not github:
+            github = Github(login_or_token=_GITHUB_ACCESS_TOKEN, timeout=GITHUB_TIMEOUT_SECONDS)
+
+        self.github = github
+        self.remaining = github.get_rate_limit().core.remaining
+
+    def _is_api_exhausted(self):
+        """Check if GH API rate limit is exhausted."""
+        self.remaining = self.github.get_rate_limit().core.remaining
```
Comment (harshad16) [inline srcopsmetrics/github_handling.py:49]: isnt this already set on the line 45 ?

**Review by harshad16 — COMMENTED** [review]
lgtm
just one question

File: srcopsmetrics/github_handling.py (hunk 1 above)
Comment (xtuchyna) [reply srcopsmetrics/github_handling.py:49]: it is, but this is just assigning it everytime we check. Without get_rate_limit() you don't really know about the _current_ remaining limit. This is then used in `iterator`

File: srcopsmetrics/iterator.py [hunk 2]
```diff
@@ -84,16 +86,13 @@ def run(self):
...
-                if remaining <= API_RATE_MINIMAL_REMAINING:
-                    self.wait_until_api_reset()
+            progressbar = tqdm(entities, total=length)
+            for idx, entity in enumerate(progressbar, 1):
+                self.knowledge_updated = True
 
-                if idx % 5 == 0:
-                    _LOGGER.info("[API requests remaining: %d]" % remaining)
+                self.handler.check_and_wait_for_api()
+                progressbar.set_postfix(ordered_dict={"RATE remaining": self.handler.remaining})
```
Comment (xtuchyna) [inline srcopsmetrics/iterator.py:95]: @harshad16 here it is constantly updated in progressbar to display the current api limit

File: srcopsmetrics/github_handling.py (hunk 1 above)
Comment (harshad16) [reply srcopsmetrics/github_handling.py:49]: ah I see, how this is being used :+1:

File: srcopsmetrics/iterator.py (hunk 2 above)
Comment (harshad16) [reply srcopsmetrics/iterator.py:95]: ack, :+1:

**Review by harshad16 — COMMENTED** [review]
/lgtm
:+1:

**Pull request merged**